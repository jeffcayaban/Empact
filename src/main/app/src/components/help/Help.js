import React from 'react';
import {Button, Col, Grid, Row} from "react-bootstrap";
import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";
import {Route} from "react-router-dom";
import {faComment, faUserAlt} from "@fortawesome/free-solid-svg-icons";
import {faClipboard} from "@fortawesome/free-solid-svg-icons/faClipboard";

/**
 * Displays the links to the different help pages for the major elements such as Petitions, Arguments and Accounts.
 */

class Help extends React.PureComponent {
    render() {
        return (
            <div>
                <div className={"centerStyle"}>
                    <h2>Help</h2>
                    <p>What do you need help with?</p>
                </div>
                <div className={"manageContentActionPanels"}>
                    <Grid>
                        <Row>
                            <Col md={4} lg={4}>
                                <div className={"centerStyle"} style={{ paddingTop: '15px' }}>
                                    <Route render={({ history }) => (
                                        <Button
                                            style={{ width: '100%', height: '100px' }}
                                            onClick={() => { history.push(`/help/argument`) }}
                                        >
                                            <div className={"centerStyle"}>
                                                <FontAwesomeIcon icon={faComment} size="3x" />
                                                <div style={{ paddingTop: '15px' }}>{'Argument'}</div>
                                            </div>
                                        </Button>
                                    )}/>
                                </div>
                            </Col>
                            <Col md={4} lg={4}>
                                <div className={"centerStyle"} style={{ paddingTop: '15px' }}>
                                    <Route render={({ history }) => (
                                        <Button
                                            style={{ width: '100%', height: '100px' }}
                                            onClick={() => { history.push(`/help/petition`) }}
                                        >
                                            <div className={"centerStyle"}>
                                                <FontAwesomeIcon icon={faClipboard} size="3x" />
                                                <div style={{ paddingTop: '15px' }}>{'Petition'}</div>
                                            </div>
                                        </Button>
                                    )}/>
                                </div>
                            </Col>
                            <Col md={4} lg={4}>
                                <div className={"centerStyle"} style={{ paddingTop: '15px' }}>
                                    <Route render={({ history }) => (
                                        <Button
                                            style={{ width: '100%', height: '100px' }}
                                            onClick={() => { history.push(`/help/account`) }}
                                        >
                                            <div className={"centerStyle"}>
                                                <FontAwesomeIcon icon={faUserAlt} size="3x" />
                                                <div style={{ paddingTop: '15px' }}>{'Account'}</div>
                                            </div>
                                        </Button>
                                    )}/>
                                </div>
                            </Col>
                        </Row>
                    </Grid>
                </div>
            </div>
        );
    }
}

export default Help;