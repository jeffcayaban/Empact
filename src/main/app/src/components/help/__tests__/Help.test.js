import React from 'react';
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";
import Help from "../Help";

describe('Help', () => {

    it('should render successfully', () => {
        const component = shallow(<Help />);
        expect(toJson(component)).toMatchSnapshot();
    });
});
