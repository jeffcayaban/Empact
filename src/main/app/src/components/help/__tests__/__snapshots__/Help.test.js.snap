// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Help should render successfully 1`] = `
<div>
  <div
    className="centerStyle"
  >
    <h2>
      Help
    </h2>
    <p>
      What do you need help with?
    </p>
  </div>
  <div
    className="manageContentActionPanels"
  >
    <Grid
      bsClass="container"
      componentClass="div"
      fluid={false}
    >
      <Row
        bsClass="row"
        componentClass="div"
      >
        <Col
          bsClass="col"
          componentClass="div"
          lg={4}
          md={4}
        >
          <div
            className="centerStyle"
            style={
              Object {
                "paddingTop": "15px",
              }
            }
          >
            <Route
              render={[Function]}
            />
          </div>
        </Col>
        <Col
          bsClass="col"
          componentClass="div"
          lg={4}
          md={4}
        >
          <div
            className="centerStyle"
            style={
              Object {
                "paddingTop": "15px",
              }
            }
          >
            <Route
              render={[Function]}
            />
          </div>
        </Col>
        <Col
          bsClass="col"
          componentClass="div"
          lg={4}
          md={4}
        >
          <div
            className="centerStyle"
            style={
              Object {
                "paddingTop": "15px",
              }
            }
          >
            <Route
              render={[Function]}
            />
          </div>
        </Col>
      </Row>
    </Grid>
  </div>
</div>
`;
